account-keys/src/impls.rs

```diff
@@ -0,0 +1,89 @@
+//! Implement mc-abstract-account-keys traits for concrete AccountKeys
+
+use crate::{AccountKeys};
+use curve25519_dalek::scalar::Scalar;
+use mc_abstract_account_keys::{RingSigner, MemoHmacSigner, KeyImageComputer, Error};
+use mc_crypto_ring_signature::{KeyImage,SignableInputRing, onetime_keys::recover_onetime_private_key};
+
+// Note: We should delete `LocalRingSigner` and just use `AccountKeys` where we were using that.
+impl RingSinger for AccountKeys {
+    fn sign(
+        &self,
+        message: &[u8],
+        ring: &SignableInputRing,
+        pseudo_output_blinding: Scalar,
+        rng: &mut dyn CryptoRngCore,
+    ) -> Result<RingMLSAG, Error> {
+        let real_input = ring
+            .members
+            .get(ring.real_input_index)
+            .ok_or(Error::RealInputIndexOutOfBounds)?;
+        let target_key = RistrettoPublic::try_from(&real_input.target_key)?;
+
+        // First, compute the one-time private key
+        let onetime_private_key = match ring.input_secret.onetime_key_derive_data {
+            OneTimeKeyDeriveData::OneTimeKey(key) => key,
+            OneTimeKeyDeriveData::SubaddressIndex(subaddress_index) => {
+                let public_key = RistrettoPublic::try_from(&real_input.public_key)?;
+
+                recover_onetime_private_key(
+                    &public_key,
+                    self.view_private_key(),
+                    &self.subaddress_spend_private(subaddress_index),
+                )
+            }
+        };
+
+        // Check if this is the correct one-time private key
+        if RistrettoPublic::from(&onetime_private_key) != target_key {
+            return Err(Error::TrueInputNotOwned);
+        }
+
+        // Note: Some implementations might be able to cache this generator
+        let generator = generators(*ring.input_secret.amount.token_id);
```

Comment:
:wrench: it was like this before so might be better to leave alone for this PR
from https://github.com/mobilecoinfoundation/mobilecoin/blob/master/CONTRIBUTING.md, https://doc.rust-lang.org/1.0.0/style/style/imports.html#prefer-fully-importing-types/traits-while-module-qualifying-functions.
```suggestion
        let generator = mc-crypto_ring_signature::generators(*ring.input_secret.amount.token_id);
```

Reply:
Eran typically comments that we should use imports at the top of the file, rather than insert module names in the middle of the file